[ryanc] iP

bipplane opened the pull request:
<h2 align="center"> ❤️ CHAEWON TASK MANAGER ❤️</h2>  

<p align="center">   
<img  
src="https://kpopping.com/documents/1a/4/800/240925-CHAEWON-Instagram-Update-documents-2(1).jpeg?v=fcd9b"  
alt="Chaewon Image"  
height="400"  
width="320">  
</p>  

<p align="center"> A simple and cute chatbot inspired by Kim Chaewon! </p>  

> Mergesort this, quicksort that, I just want to sort my life out.. 💀 

<h3> Features: </h3>  

- **Increased Productivity**: Streamline your task management.  
- **Easy to Learn**: Intuitive interface for quick adoption.  
- **Interactive**: Engaging command-driven task management.  
- **(Optional) Girl Talk**:  A fun, lighthearted theme!  
- **Task List**: Create, manage, and track your tasks.  
- **Auto-save**:  Your data is automatically saved.  
- **Reminders for Upcoming Tasks**: *(Work In Progress)*  

<h3> Getting Started: </h3>  

1.  Download the application from [here](https://github.com/bipplane/ip/).  
2.  Run the executable.  
3.  Add your tasks and let Chaewon manage them!  

<h3> Current Features (as of 2025-02-01): </h3>  

- [x] Task List Management  
- [x] Automatic Saving  
- [ ] Task Reminders *(In Development)*  

<h3> For Java Developers: </h3>  
This project can also be used for Java code-reading practice. Here's some of the `Parser` class logic:  

```java  
public Command parse(String input) {
    String[] parts = input.split(" ");
    String command = parts[0].toLowerCase();
    Command c;

    try {
        switch (command) {
        case "find", "search":
            if (parts.length < 2) {
                throw new ChaewonException("The keyword to search for cannot be empty.");
            }
            c = new FindCommand(input.split("find ")[1]);
            break;
        case "hello", "hi", "hey", "yo", "sup":
            c = new HelloCommand();
            break;
        // ... more commands
        }
    }
```
Hope you'll enjoy this little project of mine!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

bipplane committed:
Level-0: Rename, Greet, Exit

bipplane committed:
Level-0: Rename, Greet, Exit

bipplane committed:
Level-1: Echo

bipplane committed:
Level-1: Echo

bipplane committed:
Level-1: Echo

bipplane committed:
Merge tag 'refs/tags/Level-1'

bipplane committed:
Level-2: Add, List

bipplane committed:
Level-2: Add, List

bipplane committed:
Merge remote-tracking branch 'origin/master'

bipplane committed:
Level-3: Mark as Done

bipplane committed:
Level-4: ToDo, Event, Deadline

bipplane committed:
Level-4: ToDo, Event, Deadline

bipplane committed:
A-TextUiTesting

bipplane committed:
Level-5: Handle Errors

bipplane committed:
Level-6: Delete

bipplane committed:
Level-6: Delete

bipplane committed:
Added extra tests

bipplane committed:
Update README.md

bipplane committed:
Packaged tasks

bipplane committed:
Added ChaewonException

bipplane committed:
Refactor main class to use tasks package and add ChaewonException

bipplane committed:
Merge remote-tracking branch 'origin/master'

bipplane committed:
Update JavaDoc

bipplane committed:
Update output formatting

bipplane committed:
Import java.io.File

bipplane committed:
Update JavaDoc

bipplane committed:
Level-7: Save

bipplane committed:
Merge branch 'master' into branch-Level-7

bipplane committed:
Merge pull request #1 from bipplane/branch-Level-7

Branch Level 7

bipplane committed:
Update output formatting

bipplane committed:
Merge remote-tracking branch 'origin/branch-Level-7' into branch-Level-7

bipplane committed:
Merge remote-tracking branch 'origin/branch-Level-7' into branch-Level-7

bipplane committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/Chaewon.java

bipplane committed:
Merge remote-tracking branch 'origin/branch-Level-7' into branch-Level-7

bipplane committed:
Merge remote-tracking branch 'origin/branch-Level-7' into branch-Level-7

bipplane committed:
Merge pull request #2 from bipplane/branch-Level-7

Branch level 7

bipplane committed:
Merge remote-tracking branch 'origin/master'

bipplane committed:
Merge remote-tracking branch 'origin/branch-Level-7' into branch-Level-7

bipplane committed:
Merge remote-tracking branch 'origin/branch-Level-7' into branch-Level-7

bipplane committed:
Merge branch 'refs/heads/branch-Level-7'

bipplane committed:
Merge pull request #3 from bipplane/branch-Level-7

Branch level 7

bipplane committed:
Update runtest

bipplane committed:
Merge remote-tracking branch 'origin/master'

bipplane committed:
Create Ui Class

bipplane committed:
Add time format for tasks

bipplane committed:
Add time format for tasks

bipplane committed:
Merge pull request #4 from bipplane/branch-Level-8

Branch level 8

bipplane committed:
Add Ui class to deal with user interactions

bipplane committed:
Add TaskList class for task management

bipplane committed:
Add command classes for task management functionality

bipplane committed:
Add Parser class to deal with user command

bipplane committed:
Add Storage class to deal with loading & saving tasks

bipplane committed:
Add TaskList class to store list of user tasks

bipplane committed:
Add Ui class to deal with user interactions

bipplane committed:
Refactor UI messages and enhance command parsing for better user interaction

bipplane committed:
Fix file path in README

bipplane committed:
Refactor code to properly save tasks

bipplane committed:
Update tests

bipplane committed:
Merge pull request #5 from bipplane/A-MoreOOP

A-MoreOOP

bipplane committed:
Handle file creation during first startup

bipplane committed:
Merge pull request #6

A-Packages

bipplane committed:
Merge branch 'add-gradle-support'

bipplane committed:
Merge branch 'add-gradle-support'

bipplane committed:
Merge remote-tracking branch 'origin/master'

bipplane committed:
A-Gradle: Use Gradle

bipplane committed:
Add Checkstyle configuration and suppressions for coding standards

bipplane committed:
Merge pull request #7 from bipplane/A-Gradle

A-Gradle

bipplane committed:
Add unit tests for DeadlineTask and TaskList

bipplane committed:
Update build.gradle to include snapshot repository

bipplane committed:
Update build.gradle to include snapshot repository

bipplane committed:
Merge remote-tracking branch 'origin/branch-A-JUnit' into branch-A-JUnit

bipplane committed:
Add JavaFX dependencies and update shadowJar configuration

bipplane committed:
Add Javadoc comments to Task and its subclasses

bipplane committed:
Add Javadoc comments to Command and its subclasses

bipplane committed:
Add Javadoc comments to classes in package chaewon

bipplane committed:
Refactor Command class to follow Java coding standard

bipplane committed:
Merge pull request #8 from bipplane/branch-A-JavaDoc

Branch-A-JavaDoc

bipplane committed:
Merge branch 'master' into branch-A-CodingStandard

bipplane committed:
Merge pull request #9 from bipplane/branch-A-CodingStandard

Refactor Command class to follow Java coding standard

bipplane committed:
Merge pull request #9 from bipplane/branch-A-CodingStandard

Refactor Command class to follow Java coding standard

bipplane committed:
Merge remote-tracking branch 'origin/master'

bipplane committed:
Add FindCommand and task searching functionality

bipplane committed:
Merge branch 'master' into branch-Level-9

bipplane committed:
Merge pull request #10 from bipplane/branch-Level-9

Add FindCommand and task searching functionality

bipplane committed:
Merge remote-tracking branch 'origin/master'

bipplane committed:
Fix minor post-merge issues

bipplane committed:
Update README.md

bipplane committed:
Use checkstyle

bipplane committed:
Merge pull request #11 from bipplane/branch-A-CheckStyle

A-check style

bipplane committed:
Update README.md

bipplane committed:
Update README.md

bipplane committed:
Refactor gradle for GUI

bipplane committed:
Add photo for GUI

bipplane committed:
Add FXML file for GUI

bipplane committed:
Add CSS file for GUI

bipplane committed:
Add GUI components and refactor command execution methods

bipplane committed:
Fix JavaDoc

bipplane committed:
Add chatbot message delay for realism

bipplane committed:
Fix application name

bipplane committed:
Fix application not exiting after exit command

bipplane committed:
Merge pull request #12 from bipplane/branch-Level-10

Branch level 10

bipplane committed:
Enhance user interaction feature

bipplane committed:
Update README.md

bipplane committed:
Adjust UI layout

bipplane committed:
Merge remote-tracking branch 'origin/master'

LF1928 commented on src/main/java/DialogueBox.java:
```diff
@@ -0,0 +1,59 @@
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+
+/**
```
LGTM!

LF1928 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,37 @@
+import java.io.IOException;
```
package?

LF1928 commented on src/main/java/chaewon/Parser.java:
```diff
@@ -0,0 +1,89 @@
+
+        try {
+            switch (command) {
+            case "find", "search":
+                if (parts.length < 2) {
+                    throw new ChaewonException("The keyword to search for cannot be empty.");
+                }
+                c = new FindCommand(input.split("find ")[1]);
+                break;
+            case "hello", "hi", "hey", "yo", "sup":
+                c = new HelloCommand();
+                break;
+            case "bye", "goodbye", "exit", "quit", "cya":
+                c = new ExitCommand();
+                break;
+            case "todo", "task":
+                if (parts.length < 2) {
+                    throw new ChaewonException("The description of a todo cannot be empty.");
+                }
+                c = new TodoCommand(input.split("todo ")[1]);
+                break;
+            case "deadline":
+                if (parts.length < 2) {
+                    throw new ChaewonException("The description of a deadline cannot be empty.");
+                }
+                c = new DeadlineCommand(input.split("deadline ")[1]);
+                break;
+            case "event":
+                if (parts.length < 2) {
+                    throw new ChaewonException("The description of an event cannot be empty.");
+                }
+                c = new EventCommand(input.split("event ")[1]);
+                break;
```
I like how you had more command words!

LF1928 reviewed (commented):
Overall, I find your code organised and easy to read maybe saveTasks can be done in another way? but otherwise looks good!

bipplane committed:
Add assertions to document important assumptions

bipplane committed:
Use streams to chain tasks

bipplane committed:
Enhance code quality and readability

bipplane committed:
Merge pull request #13 from bipplane/branch-A-Assertions

Add assertions to document important assumptions

bipplane committed:
Merge pull request #14 from bipplane/branch-A-CodeQuality

Enhance code quality and readability

bipplane committed:
Merge branch 'master' into branch-A-Streams

bipplane committed:
Merge pull request #15 from bipplane/branch-A-Streams

Streams

bipplane committed:
Add GitHub Actions workflow for Java CI with Gradle

There is no Continuous Integration (CI) in this Java project.

Commits to a PR may cause build or test failures against the default branch.

Let's add this .yml file to allow such problems to be observed, ensuring code functionality.

Refer to this article for more info
https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-java-with-gradle

bipplane committed:
Merge pull request #16 from bipplane/branch-A-CI

Add GitHub Actions workflow for Java CI with Gradle

There is no Continuous Integration (CI) in this Java project.

Commits to a PR may cause build or test failures against the default branch.

Let's add this .yml file to allow such problems to be observed, ensuring code functionality.

Refer to this article for more info
https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-java-with-gradle

bipplane committed:
Add conflict detection for EventTask and corresponding tests

Events with clashing timings are allowed to be added without warning.

This is not ideal for a proper schedule management for our users.

Let's add a function that checks for such clashes to prevent this.

This will warn users about clashing tasks for increased QOL.

bipplane committed:
Refactor package structure and update main class references to 'gui'

bipplane committed:
Improve GUI and chatbot personality

bipplane committed:
Update user guide and enhance UI styles

bipplane committed:
Update README.md

bipplane committed:
Update user guide and enhance UI styles

bipplane committed:
Adjust font sizing
